src/cls/tabular/fbwriter_fbu.cc

```diff
@@ -0,0 +1,932 @@
+          auto fv = Tables::CreateSDT_FLOAT_FBU( builder, float_vect_fb ) ;
+          data_vect.push_back( fv.Union() ) ;
+        }
+        else if( boost::algorithm::ends_with( key, std::to_string( Tables::SDT_STRING ) ) ) {
+          std::vector< flatbuffers::Offset<flatbuffers::String> > single_sv ;
+          single_sv.push_back( filedata.listof_string_vect_raw[ std::stoi( index ) ][ i ] ) ;
+          auto string_vect_fb = builder.CreateVector( single_sv ) ;
+          auto sv = Tables::CreateSDT_STRING_FBU( builder, string_vect_fb ) ;
+          data_vect.push_back( sv.Union() ) ;
+        }
+        else {
+          std::cout << ">>> unrecognized key '" << key << "'" << std::endl ;
+          exit( 1 ) ;
+        }
+      } // for loop : creates one data_vect row
+
+      auto data = builder.CreateVector( data_vect ) ;
+      auto rec = Tables::CreateRecord_FBU(
+        builder,                 //builder ptr
+        i,                       //rid
+        nullbits_vector_fb,      //nullbits vect
+        record_data_types,       //data types vect
+        data ) ;                 //data vect
+      row_records.push_back( rec ) ;
+
+    } // for loop : processes and saves all rows
+
+    auto row_records_fb = builder.CreateVector( row_records ) ;
+
+    // create the Rows flatbuffer:
+    auto rows = Tables::CreateRows_FBU(
+      builder,            //builder
+      row_records_fb ) ;  //data
+
+    // generate schema string
+    std::string schema_string = "" ;
+    for( unsigned int i = 0; i < ncols; i++ ) {
+      std::string att_name = schema_attnames[ i ] ;
+      auto att_type = schema_datatypes_sdt[ i ] ;
+      std::string this_entry = " " + std::to_string( i ) + " " ;
+      switch( att_type ) {
+        case Tables::SDT_UINT64 :
+          this_entry = this_entry + std::to_string( Tables::SDT_UINT64 ) ;
+          break ;
+        case Tables::SDT_FLOAT :
+          this_entry = this_entry + std::to_string( Tables::SDT_FLOAT ) ;
+          break ;
+        case Tables::SDT_STRING :
+          this_entry = this_entry + std::to_string( Tables::SDT_STRING ) ;
+          break ;
+        default :
+          std::cout << ">>4 unrecognized att_type '" << att_type << "'" << std::endl ;
+          exit( 1 ) ;
+      } //switch
+      this_entry = this_entry + " 0 0 " + att_name + " \n" ;
+      schema_string = schema_string + this_entry ;
+    } //for
+
+    auto schema_string_fb = builder.CreateString( schema_string ) ;
+    if( debug )
+      std::cout << "schema_string = " << schema_string << std::endl ;
+
+    auto db_schema_name = builder.CreateString( "kats_test" ) ;
+    std::vector< uint8_t > delete_vector ( nrows, 0 ) ; //initialize with one 0 per row.
+    auto delete_vector_fb = builder.CreateVector( delete_vector ) ;
+
+    auto root = CreateRoot_FBU(
+      builder,                          //builder
+      SFT_FLATBUF_UNION_ROW,           //data_format_type
+      0,                                //skyhook_version
```

Comment:
We should probably use the const values here if you can, like you did in the other file I believe.